drv/stm32g0-rcc/src/main.rs

```diff
@@ -0,0 +1,213 @@
+//! above.
+//!
+//! ## `disable_clock` (2)
+//!
+//! Requests that the clock to a peripheral be turned off.
+//!
+//! Request message format: single `u32` giving peripheral index as described
+//! for `enable_clock`.
+//!
+//! ## `enter_reset` (3)
+//!
+//! Requests that the reset line to a peripheral be asserted.
+//!
+//! Peripherals are numbered by bit number in the RCC control registers, as
+//! follows:
+//!
+//! - RCC_IOPRSTR[31:0] are indices 31-0.
+//! - RCC_AHBRSTR[31:0] are indices 63-32.
+//! - RCC_APBRSTR1[31:0] are indices 95-64.
+//! - RCC_APBRSTR2[31:0] are indices 127-96.
+//!
+//! Request message format: single `u32` giving peripheral index as described
+//! above.
+//!
+//! ## `leave_reset` (4)
+//!
+//! Requests that the reset line to a peripheral be deasserted.
+//!
+//! Request message format: single `u32` giving peripheral index as described
+//! for `enter_reset`.
+
+#![no_std]
+#![no_main]
+
+#[cfg(feature = "g031")]
+use stm32g0::stm32g031 as device;
+
+#[cfg(feature = "g070")]
+use stm32g0::stm32g070 as device;
+
+#[cfg(feature = "g0b1")]
+use stm32g0::stm32g0b1 as device;
+
+use idol_runtime::RequestError;
+use userlib::*;
+
+#[derive(Copy, Clone, Debug, FromPrimitive)]
+#[repr(u32)]
+pub enum RccError {
+    NoSuchPeripheral = 1,
+}
+
+impl From<u32> for RccError {
+    fn from(x: u32) -> Self {
+        match x {
+            1 => RccError::NoSuchPeripheral,
+            _ => panic!(),
+        }
+    }
+}
+
+impl From<RccError> for u16 {
+    fn from(x: RccError) -> Self {
+        x as u16
+    }
+}
+
+#[derive(FromPrimitive)]
+enum Bus {
+    Iop = 0,
+    Ahb = 1,
+    Apb1 = 2,
+    Apb2 = 3,
+}
+
+// None of the registers we interact with have the same types, and they share no
+// useful traits, so we can't extract the bit-setting routine into a function --
+// we have no choice but to use macros.
+macro_rules! set_bits {
+    ($reg:expr, $mask:expr) => {
+        $reg.modify(|r, w| unsafe { w.bits(r.bits() | $mask) })
+    };
+}
+
+// None of the registers we interact with have the same types, and they share no
+// useful traits, so we can't extract the bit-clearing routine into a function
+// -- we have no choice but to use macros.
+macro_rules! clear_bits {
+    ($reg:expr, $mask:expr) => {
+        $reg.modify(|r, w| unsafe { w.bits(r.bits() & !$mask) })
+    };
+}
+
+#[export_name = "main"]
+fn main() -> ! {
+    // From thin air, pluck a pointer to the RCC register block.
+    //
+    // Safety: this is needlessly unsafe in the API. The RCC is essentially a
+    // static, and we access it through a & reference so aliasing is not a
+    // concern. Were it literally a static, we could just reference it.
+    let rcc = unsafe { &*device::RCC::ptr() };
+
+    // Any global setup we required would go here.
+
+    // Field messages.
+    // Ensure our buffer is aligned properly for a u32 by declaring it as one.
+    let mut buffer = [0u8; idl::INCOMING_SIZE];
```

Comment:
...this doesn't look like a `u32`, contrary to the comment above

(same issue exists in the H7 RCC task)

Reply:
That commit is obsolete -- I should have removed it when I did the stm32h7 rcc Idol conversion, and it looks like it got copied into the stm32g0 version. Will fix.